mycroft/util/lang/format_de.py

```diff
@@ -0,0 +1,344 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from mycroft.util.lang.format_common import convert_to_mixed_fraction
+from math import floor
+
+NUM_STRING_DE = {
+    0: 'null',
+    1: 'ein',  # ein Viertel etc., nicht eins Viertel
+    2: 'zwei',
+    3: 'drei',
+    4: 'vier',
+    5: u'fünf',
+    6: 'sechs',
+    7: 'sieben',
+    8: 'acht',
+    9: 'neun',
+    10: 'zehn',
+    11: 'elf',
+    12: u'zwölf',
+    13: 'dreizehn',
+    14: 'vierzehn',
+    15: u'fünfzehn',
+    16: 'sechzehn',
+    17: 'siebzehn',
+    18: 'achtzehn',
+    19: 'neunzehn',
+    20: 'zwanzig',
+    30: u'dreißig',
+    40: 'vierzig',
+    50: u'fünfzig',
+    60: 'sechzig',
+    70: 'siebzig',
+    80: 'achtzig',
+    90: 'neunzig',
+    100: 'hundert'
+}
+
+# German uses "long scale" https://en.wikipedia.org/wiki/Long_and_short_scales
+# Currently, numbers are limited to 1000000000000000000000000,
+# but NUM_POWERS_OF_TEN can be extended to include additional number words
+
+
+NUM_POWERS_OF_TEN = [
+    '', 'tausend', 'Million', 'Milliarde', 'Billion', 'Billiarde', 'Trillion',
+    'Trilliarde'
+]
+
+FRACTION_STRING_DE = {
+    2: 'halb',
+    3: 'drittel',
+    4: 'viertel',
+    5: u'fünftel',
```

Comment:
Not an issue just a general comment: The unicode `u` shouldn't be needed anymore, since the switch to python3 all strings are unicode :)

Reply:
Okay, I'll keep it in mind

Reply:
the blank lines should all be gone now